src/pages/api/auth/[...nextauth].ts

```diff
@@ -0,0 +1,79 @@
+import NextAuth from 'next-auth';
+import CredentialsProvider from 'next-auth/providers/credentials';
+import { userSchema } from 'src/api/models/user';
+import * as z from 'zod';
+
+const authInfoDtoSchema = z.object({
+  id: z.string(),
+  username: z.string(),
+  permissions: z.array(z.enum(['listread', 'listwrite'])),
+});
+
+const permissionsMap = {
+  listread: 'read',
+  listwrite: 'write',
+} as const;
+
+const credentialsSchema = z.object({
+  token: z.string().min(1),
+});
+
+export default NextAuth({
+  providers: [
+    CredentialsProvider({
+      name: 'Credentials',
+      credentials: {
+        token: { type: 'text' },
+      },
+      async authorize(credentials, _req) {
+        const { token } = credentialsSchema.parse(credentials);
+
+        const response = await fetch('https://api.vndb.org/kana/authinfo', {
+          method: 'GET',
+          headers: {
+            Authorization: `Token ${token}`,
+          },
+        });
+
+        if (response.ok) {
+          const authInfoDto = await response.json();
+          const authInfo = authInfoDtoSchema.parse(authInfoDto);
+
+          return {
+            user: {
+              id: authInfo.id,
+              token,
+              name: authInfo.username,
+              permissions: authInfo.permissions.map(permission => permissionsMap[permission]),
+            },
+          };
+        }
```

Comment:
Feels like we need to create proper mapper for this. But I don't think we will gain anything. Especially if would need to refactor auth to creds, mappers and models will go to waste. 
@Moltenship What do you think?

Reply:
After checking whole PR, I think we should create a mapper and `User` and `UserPermission` models, because they're reused in other places (`next-auth.d.ts`).